micro-rdk-installer/src/main.rs

```diff
@@ -331,3 +367,105 @@ fn main() -> Result<(), Error> {
     };
     Ok(())
 }
+
+fn update_app_image(args: &AppImageArgs) -> Result<(), Error> {
+    let config = Config::load().map_err(|err| Error::SerialConfigError(err.to_string()))?;
+
+    let dir = tempfile::tempdir().map_err(Error::FileError)?;
+    let tmp_old = dir.path().join("running-ptable.img");
+
+    log::info!("Retrieving running partition table");
+    let mut flasher =
+        connect(&args.connect_args, &config, false, false).map_err(|_| Error::FlashConnect)?;
+    flasher
+        .read_flash(
+            PARTITION_TABLE_ADDR.into(),
+            PARTITION_TABLE_SIZE.into(),
+            0x1000,
+            64,
+            tmp_old.clone(),
+        )
+        .map_err(|_| Error::FlashConnect)?;
+
+    let old_ptable_buf = fs::read(tmp_old).map_err(Error::FileError)?;
+    let old_ptable = PartitionTable::try_from_bytes(old_ptable_buf.clone())
+        .map_err(|e| Error::PartitionTableError(e.to_string()))?;
+
+    log::info!("Retrieving new image");
+    let tmp_new = tempfile::Builder::new()
+        .prefix("micro-rdk-bin")
+        .tempdir()
+        .map_err(Error::FileError)?;
+    let app_path_new = match &args.flash_args.bootloader {
+        Some(path) => PathBuf::from(path),
+        None => {
+            let rt = Runtime::new().map_err(Error::AsyncError)?;
+            rt.block_on(download_micro_rdk_release(&tmp_new, args.version.clone()))?
+        }
+    };
+
+    let mut new_ptable_buf = vec![0; PARTITION_TABLE_SIZE.into()];
+    log::info!("Extracting new partition table");
+    let mut app_file_new = OpenOptions::new()
+        .read(true)
+        .write(true)
+        .open(app_path_new.clone())
+        .map_err(Error::FileError)?;
+
+    let file_len = app_file_new.metadata().map_err(Error::FileError)?.len();
+    if file_len < (PARTITION_TABLE_ADDR as u64 + PARTITION_TABLE_SIZE as u64) {
+        return Err(Error::PartitionTableError(
+            "file length is less than partition size".to_string(),
+        ));
+    }
+
+    let _ = app_file_new
+        .seek(SeekFrom::Start(PARTITION_TABLE_ADDR.into()))
+        .map_err(Error::FileError)?;
+    app_file_new
+        .read_exact(&mut new_ptable_buf)
+        .map_err(Error::FileError)?;
+    let new_ptable = PartitionTable::try_from_bytes(new_ptable_buf.clone())
+        .map_err(|e| Error::PartitionTableError(e.to_string()))?;
+
+    // Compare partition tables
+    if old_ptable != new_ptable {
+        log::error!("rebuild and flash micro-rdk from scratch");
+        return Err(Error::PartitionTableError(
+            "old and new app image partition tables do not match".to_string(),
+        ));
+    }
+
+    log::info!("Partition tables match!");
```

Comment:
If you are logging/erroring if they don't match, you probably don't need to log if they do